sys/netinet/in.c

```diff
@@ -452,7 +452,7 @@ in_aifaddr_ioctl(u_long cmd, caddr_t data, struct ifnet *ifp, struct thread *td)
 	if (ifp->if_flags & IFF_POINTOPOINT)
 		ia->ia_dstaddr = *dstaddr;
 
-	if (vhid != 0) {
+	if (vhid != 0 && carp_attach_p != NULL) {
```

Comment:
So why isn't the check at line 380 enough to keep this from happening?

Reply:
https://github.com/freebsd/freebsd-src/blob/80b1995bb5d6e4a962c9a53ddede2420c299d9e4/sys/netinet/in.c#L379-L380

In our case `vhid` is negative

Reply:
In other places, we reject as invalide vhids < 0...  I think that someone more familiar with carp than I am is going to look at this...

Reply:
Sounds like a plan, thanks! I hope the repro's going to be of use.

Reply:
I asked Kristof to take a look. His question was why not change the above to vhid != 0? It's not valid to set vhid to any value when carp isn't loaded.

Reply:
Thanks, great suggestion. Implemented it.